akka-stream/src/main/scala/akka/stream/scaladsl/Source.scala

```diff
@@ -472,9 +479,108 @@ object Source {
    *
    * @see [[Source.lazily]]
    */
+  @deprecated("Use 'Source.lazyFuture' instead", "2.6.0")
   def lazilyAsync[T](create: () => Future[T]): Source[T, Future[NotUsed]] =
     lazily(() => fromFuture(create()))
 
+  /**
+   * Emits a single value when the given `Future` is successfully completed and then completes the stream.
+   * The stream fails if the `Future` is completed with a failure.
+   */
+  def future[T](futureElement: Future[T]): Source[T, NotUsed] =
+    fromGraph(new FutureSource[T](futureElement))
+
+  /**
+   * Emits a single value when the given `CompletionStage` is successfully completed and then completes the stream.
+   * If the `CompletionStage` is completed with a failure the stream is failed.
+   *
+   * Here for Java interoperability, the normal use from Scala should be [[Source.future]]
+   */
+  def completionStage[T](completionStage: CompletionStage[T]): Source[T, NotUsed] =
+    future(completionStage.toScala)
+
+  /**
+   * Turn a `Future[Source]` into a source that will emit the values of the source when the future completes successfully.
+   * If the `Future` is completed with a failure the stream is failed.
+   */
+  def futureSource[T, M](futureSource: Future[Source[T, M]]): Source[T, Future[M]] =
+    fromGraph(new FutureFlattenSource(futureSource))
+
+  /**
+   * Defers invoking the `create` function to create a single element until there is downstream demand.
+   *
+   * If the `create` function fails when invoked the stream is failed.
+   *
+   * Note that asynchronous boundaries (and other operators) in the stream may do pre-fetching which counter acts
+   * the laziness and will trigger the factory immediately.
+   *
+   * The materialized future `Done` value is completed when the `create` function has successfully been invoked,
+   * if the function throws the future materialized value is failed with that exception.
+   * If downstream cancels or fails before the function is invoked the materialized value
+   * is failed with a [[akka.stream.NeverMaterializedException]]
+   */
+  def lazySingle[T](create: () => T): Source[T, Future[Done]] =
```

Comment:
I'd vote for removing the materialized value. Isn't it an anti-pattern to attach some completion signal to a source? In the end it doesn't mean anything but that the function has been invoked (which the user can figure out itself because it provides the function).

Reply:
Same for `lazyFuture` and the completetionStage variants in javadsl.